packages/entities/entities-shared/src/components/entity-link/EntityLink.vue

```diff
@@ -0,0 +1,134 @@
+<template>
+  <div
+    class="kong-ui-public-entity-link"
+    data-testid="entity-link-parent"
+  >
+    <div
+      v-if="isDeletedEntity"
+      class="deleted-entity"
+    >
+      {{ deletedUuidDisplay }}
+    </div>
+    <div v-else-if="!entityUuid">
+      {{ ' – ' }}
+    </div>
+    <KExternalLink
+      v-else-if="entityUuid"
+      class="entity-link"
+      :href="externalLink"
+      target="_blank"
+    >
+      <KTooltip
+        :label="isTruncated && entityLinkData.label || ''"
+      >
+        <span
+          ref="textContent"
+          class="entity-link-label"
+        >
+          {{ entityLinkData.label }}
+        </span>
+      </KTooltip>
+    </KExternalLink>
+    <CopyUuid
+      v-if="entityUuid"
+      class="entity-link-copy-id"
+      data-testid="copy-id"
+      format="hidden"
+      :icon-color="KUI_COLOR_TEXT_PRIMARY"
+      :notify="notifyCopy"
+      :tooltip="t('global.actions.copyId')"
+      :use-mono="false"
+      :uuid="entityUuid"
+    />
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, ref } from 'vue'
+import type { PropType, Ref } from 'vue'
+import type { CopyUuidNotifyParam } from '@kong-ui-public/copy-uuid'
+import type { EntityLinkData } from '../../types'
+import composables from '../../composables'
+import { CopyUuid } from '@kong-ui-public/copy-uuid'
+import { KUI_COLOR_TEXT_PRIMARY } from '@kong/design-tokens'
+
+const props = defineProps({
+  entityLinkData: {
+    type: Object as PropType<EntityLinkData>,
+    required: true,
+  },
+  externalLink: {
+    type: String,
+    required: true,
+  },
+})
+
+const textContent = ref<HTMLElement>()
+const { isTruncated } = composables.useTruncationDetector(textContent as Ref<HTMLElement>)
+
+const { i18n: { t } } = composables.useI18n()
+const { notify } = composables.useToaster()
```

Comment:
question: can we move away from using a toaster here? 

I would think just changing the tooltip text on click would be enough, or this component could just emit an event to the host app.

Reply:
If we emit an event to the host app, then we end up with duplicated code in multiple places in the consuming app...

I prefer the tooltip text change on click, if this is an established pattern

Reply:
Yes, this is the pattern we use for copy events, etc.